phpBB/includes/acp/acp_storage.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+/**
+* @ignore
+*/
+if (!defined('IN_PHPBB'))
+{
+	exit;
+}
+
+class acp_storage
+{
+	/** @var \phpbb\config $config */
+	protected $config;
+
+	/** @var \phpbb\language\language $lang */
+	protected $lang;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var \phpbb\di\service_collection */
+	protected $provider_collection;
+
+	/** @var \phpbb\di\service_collection */
+	protected $storage_collection;
+
+	/** @var string */
+	public $page_title;
+
+	/** @var string */
+	public $tpl_name;
+
+	/** @var string */
+	public $u_action;
+
+	public function main($id, $mode)
+	{
+		global $phpbb_container, $phpbb_dispatcher;
+
+		$this->config = $phpbb_container->get('config');
+		$this->lang = $phpbb_container->get('language');
+		$this->request = $phpbb_container->get('request');
+		$this->template = $phpbb_container->get('template');
+		$this->user = $phpbb_container->get('user');
+		$this->provider_collection = $phpbb_container->get('storage.provider_collection');
+		$this->storage_collection = $phpbb_container->get('storage.storage_collection');
+
+		// Add necesary language files
+		$this->lang->add_lang(array('common'));
+		$this->lang->add_lang(array('acp/storage'));
+
+		/**
+		 * Add language strings
+		 *
+		 * @event core.acp_storage_load
+		 * @since 3.3.0-a1
+		 */
+		$phpbb_dispatcher->dispatch('core.acp_storage_load');
+
+		$this->overview($id, $mode);
+	}
+
+	public function overview($id, $mode)
+	{
+		$form_name = 'acp_storage';
+		add_form_key($form_name);
```

Comment:
You're adding a form key but you're never actually checking it. This is however vital for the CSRF protection.